app/src/main/java/com/epfl/drawyourpath/database/FirebaseDatabaseUtils.kt

```diff
@@ -207,4 +267,52 @@ object FirebaseDatabaseUtils {
             chatList = transformChatList(data.child(FirebaseKeys.USER_CHATS)),
         )
     }
+
+    /**
+     * Helper function to convert a data snapshot to a tournament
+     * @param data the data snapshot containing the tournament
+     * @return the tournament, or null if an error occurred
```

Comment:
Suggestion: Consider using an exception instead of returning null to indicate errors